### ISSUE TITLE
R package to retrieve NASA datasets 🛰

### ISSUE THREAD

**Issue by jimhester** [ropensci/unconf17#67]
Basically an R port of https://github.com/bmtgoncalves/pyNASA, seems a pretty natural fit for rOpenSci

**Comment by stefaniebutland** [comment]
Calling all #NASADatanauts! @jasdumas @ldecicco-USGS @AmeliaMN @eringrand @LucyMcGowan

**Comment by LucyMcGowan** [comment]
yes! this has been on my todo list probably pulling from here: https://data.nasa.gov/browse?limitTo=datasets 🚀

**Comment by eringrand** [comment]
Yes!

Apologies for any typos.

**Comment by ldecicco-USGS** [comment]
Yeah, I've been looking for a way to contribute, and this is a natural fit. I've led a similar effort for USGS data (sorry for the non-unique package name...):

https://github.com/USGS-R/dataRetrieval

I can try to contribute my experience of what has worked well in that package (...and what didn't!)

**Comment by d4tagirl** [comment]
I'm not going to the Unconf but I've been talking about collaborate with this with @bmtgoncalves. Is there a way that I can be involved?

**Comment by stefaniebutland** [comment]
@d4tagirl A couple of people have participated in the unconf remotely in the past. The key would be good communication with the project team, using something like https://appear.in/ for video chat.

**Comment by d4tagirl** [comment]
I'll stay tuned then, thank you!!

**Comment by alisonboyer** [comment]
I am on board! I work at a NASA data center (https://daac.ornl.gov/ ) and we could start here for the Earth science data:
https://cmr.earthdata.nasa.gov/opensearch/
I would also like to talk about packages to retrieve actual data (as opposed to metadata) using webservices such as OPeNDAP. we could start here: https://thredds.daac.ornl.gov/thredds/catalogs/ornldaac/ornldaac.html

**Comment by cvitolo** [comment]
Count me in for this! @alisonboyer, I agree retrieving actual data would be even more interesting

**Comment by maelle** [comment]
I had started working on [this package](https://github.com/maelle/laads) interfacing the NASA API of MODIS Level 1 and Atmosphere data products. Not sure how much it overlaps with other datasets presented here, maybe not at all? In any case, if anyone is interested, I can move it to ropenscilabs for the unconf.

**Comment by alisonboyer** [comment]
@maelle Your LAADS package looks cool. This is exactly the type of package that we need. The MODISTools package was my inspiration, and I think we need multiple packages to hit different data.

**Comment by maelle** [comment]
Cool, happy to collaborate!  MODISTools has a slightly misleading name IMO since it doesn't hit all MODIS data 😊

**Comment by mdsumner** [comment]
We use `raadsync` to download data from a variety of sources via different methods (mostly wget) and includes remote sensing files from the NASA ocean colour website: 

https://github.com/AustralianAntarcticDataCentre/raadsync

It's not polished sufficiently for general use but it's not hard to set up and it's proven to work very well  -
 we'd appreciate any feedback or interest in it.  We register datasets in JSON with the information required about how to discover new/changed files, the download method, whether unpacking is required after download.

**Comment by sckott** [comment]
@alisonboyer regarding opendap, been trying to get an opendap client `dappr` https://github.com/ropensci/dappr to a useable state for a while - could be part of this

**Comment by alisonboyer** [comment]
@sckott nice! yes, that is a good idea

**Comment by alisonboyer** [comment]
i'm so sorry to miss the unconf this year. If you all decide to pursue this topic will you loop me in via video chat? I really would love to contribute

**Comment by bmtgoncalves** [comment]
Hi everyone (and @jimhester and @d4tagirl in particular)

Thank you for the interest in pyNASA. While I'm fairly comfortable using R, I've never tried writing a package. If there's anything I can do to help, just let me know!

Best

B
